main.go

```diff
@@ -223,3 +224,29 @@ func makeHTTPRequestHandler(watchdogConfig config.WatchdogConfig) func(http.Resp
 
 	}
 }
+
+func lockFilePresent() bool {
+	path := filepath.Join(os.TempDir(), ".lock")
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
+
+func makeHealthHandler() func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Should this not be the same as the old watchdog implementation? The `acceptingConnections` check is missing here and from the `lockFilePresent()`. Why is that no longer needed?

Reply:
The `acceptingConnections` check is part of the graceful shutdown that's currently only in the original watchdog. There's an issue to port it over here: #10 

I'm starting to work on that. Hoping to have a PR tonight (US time)